src/main/scala/strawman/collection/Iterator.scala

```diff
@@ -383,6 +383,42 @@ trait Iterator[+A] extends IterableOnce[A] { self =>
   }
 
   final def size: Int = length
+  
+  def intersperse[B >: A](sep: B): Iterator[B] = new Iterator[B] {
+    var intersperseNext = false
+    override def hasNext = intersperseNext || self.hasNext
+    override def next() = {
+      val n = if (intersperseNext) sep else self.next()
+      intersperseNext = !intersperseNext && self.hasNext
+      n
+    }
+  }
+  
+  def intersperse[B >: A](start: B, sep: B, end: B): Iterator[B] = new Iterator[B] {
```

Comment:
@julienrf How can I test the performance of this?

Reply:
You can have a look at the benchmarks/time project and follow the same pattern. Create two benchmark classes with one benchmark method that has the same name (e.g. `intersperse`). In one class, call the the `intersperse` implementation based on combining `View`s, in the other class, call the `intersperse` implemntation implemented on `Iterator`. Does that make sense? I won’t have time to write more documentation this week.